verifier/src/evidence.rs

```diff
@@ -199,47 +203,133 @@ where
             certificate_verifier,
             _trusted_identities: trusted_identities.into_iter().map(Into::into).collect(),
             time,
+            tcb_verifier: Cell::new(None),
         }
     }
 
+    fn verify_certificate_chain<'c, 'd>(
+        &self,
+        chain: &[Certificate],
+        crls: impl IntoIterator<Item = &'d CertificateList>,
+    ) -> (
+        VerifyingKey,
+        VerificationOutput<Option<CertificateChainVerifierError>>,
+    ) {
+        let result = self
+            .certificate_verifier
+            .verify_certificate_chain(chain, crls, self.time);
+        let is_success = result.is_ok() as u8;
+
+        // Using the default key will result in the user seeing "Error verifying the signature" for
+        // the signed data. So we try to get the key from the certificate chain, even if the
+        // verification failed. This handles the most likely failure case of an expired
+        // certificate, whose key is still the key that signed the data of interest.
```

Comment:
Let me know if this needs more explanation. Or if there are concerns with this approach

Reply:
I added a comment with links back to the docs for certificate chain order assumption.

I think IFF this assumed invariant is broken we need to we can put a normalizer on `Collateral` that ensures this order.